packages/cosmos_ui_components/lib/components/gradient_avatar.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:flutter/material.dart';
+import 'package:string_to_hex/string_to_hex.dart';
+
+class GradientAvatar extends StatelessWidget {
+  final String stringKey;
+
+  const GradientAvatar({Key? key, required this.stringKey}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final colors = generateColorsFromAddress(stringKey);
+    return CircleAvatar(
+      child: Container(
+        decoration: BoxDecoration(
+          shape: BoxShape.circle,
+          gradient: LinearGradient(colors: [colors[0], colors[1]]),
+        ),
+      ),
+    );
+  }
+
+  List<Color> generateColorsFromAddress(String address) {
+    final color1 = StringBuffer();
+    final color2 = StringBuffer();
+
+    color1.write('#');
+    color2.write('#');
+
+    // Initialised it here because linter was warning me not to declare multiple variables in single line
+    var j = address.length - 7;
+    for (var i = address.length - 1; color1.length <= 7 && color2.length <= 7; i--, j--) {
```

Comment:
this is a bit strange construction of for loop with two separate iterators and condition not depending on the iterator itself. a `while` loop here would have more sense. also we're not handling the edge cases where the address might be shorter than 14 letters. how about we do something like:

```dart
    final color1 = address.substring(0, address.length ~/ 2 + 1);
    final color2 = address.substring(address.length ~/ 2 + 1);
    return [
      Color(StringToHex.toColor(color1)),
      Color(StringToHex.toColor(color2)),
    ];
```
but it will only make sense if web does the same thing.